service/worker/scheduler/workflow_test.go

```diff
@@ -1311,6 +1320,140 @@ func (s *workflowSuite) TestBackfillInclusiveStartEnd() {
 	)
 }
 
+func (s *workflowSuite) TestHugeBackfillAllowAll() {
```

Comment:
What is the expected behavior tested by this test?

Reply:
When backfillRuns exceeds the max buffer size, the new logic will successfully run all the requested workflows, the old logic will fail (due to hitting the buffer size).

In the other test (line 1390) I reduce the buffer size to make the test more meaningful, I should do the same here